Userland/Libraries/LibWeb/Layout/LayoutState.cpp

```diff
@@ -354,9 +354,9 @@ AvailableSpace LayoutState::UsedValues::available_inner_space_or_constraints_fro
     auto inner_width = available_width_inside();
     auto inner_height = available_height_inside();
 
-    if (inner_width.is_indefinite() && outer_space.width.is_intrinsic_sizing_constraint())
+    if ((inner_width.is_indefinite() || !isfinite(inner_width.to_px().value())) && outer_space.width.is_intrinsic_sizing_constraint())
         inner_width = outer_space.width;
-    if (inner_height.is_indefinite() && outer_space.height.is_intrinsic_sizing_constraint())
+    if ((inner_height.is_indefinite() || !isfinite(inner_height.to_px().value())) && outer_space.height.is_intrinsic_sizing_constraint())
         inner_height = outer_space.height;
```

Comment:
since this change affects all formatting contexts I suggest to move this change into separate commit with description why this additional check is needed.

Reply:
Done. I'm not sure if it currently affects any other formatting context, it requires a pretty particular set of circumstances (nested tables, basically).